src/layout/Summary2/SummaryComponent2/PageSummary.tsx

```diff
@@ -1,27 +1,24 @@
 import React from 'react';
 
 import { ComponentSummary } from 'src/layout/Summary2/SummaryComponent2/ComponentSummary';
-import { useGetPage } from 'src/utils/layout/NodesContext';
 import { useNodeTraversal } from 'src/utils/layout/useNodeTraversal';
-import type { CompInternal } from 'src/layout/layout';
 
 interface PageSummaryProps {
   pageId: string;
-  summaryOverrides: CompInternal<'Summary2'>['overrides'];
 }
 
-export function PageSummary({ pageId, summaryOverrides }: PageSummaryProps) {
-  const page = useGetPage(pageId);
-  const children = useNodeTraversal((t) => (t.targetIsPage() ? t.children() : undefined), page);
-  if (!page || !children) {
+export function PageSummary({ pageId }: PageSummaryProps) {
+  const page = useNodeTraversal((dings) => dings.findPage(pageId));
+  const children = useNodeTraversal((t) => t.children(), page);
+
+  if (!page) {
     throw new Error('PageId invalid in PageSummary.');
   }
 
-  return children.map((child) => (
+  return children?.map((child, idx) => (
     <ComponentSummary
       componentNode={child}
-      key={child.id}
-      summaryOverrides={summaryOverrides}
+      key={idx}
```

Comment:
The `child.id` might be a better key since it now should be unique after the performance PR.

Reply:
hmm but is that guaranteed? It wont actually swap the ID will it,  just throw a warning?

Reply:
That is true, it might not be guaranteed. 🤔 Maybe just slap the index to the end of the id? 🤷 Atleast then it is more unique than just the index.